src/Microsoft.SqlTools.ServiceLayer/ObjectExplorer/ObjectExplorerService.cs

```diff
@@ -543,25 +547,27 @@ private async Task<ConnectionCompleteParams> Connect(ConnectParams connectParams
                 }
                 else
                 {
-                    await SendSessionFailedNotification(uri, result.ErrorMessage);
+                    await SendSessionFailedNotification(uri, result.ErrorMessage, result.ErrorNumber);
                     return null;
                 }
 
             }
             catch (Exception ex)
             {
-                await SendSessionFailedNotification(uri, ex.ToString());
+                int? errorNum = ex is SqlException sqlEx ? sqlEx.Number : null;
+                await SendSessionFailedNotification(uri, ex.ToString(), errorNum);
                 return null;
             }
         }
 
-        private async Task SendSessionFailedNotification(string uri, string errorMessage)
+        private async Task SendSessionFailedNotification(string uri, string errorMessage, int? errorCode = null)

```

Comment:
I'd suggest removing the default here so there's less chance of someone forgetting to add that parameter when making changes in the future.